### ISSUE TITLE
Cross Validation: Able to fit() but unable to refit()

### ISSUE THREAD

**Issue by tanchangsheng** [automl/auto-sklearn#301]
Was able to fit the model successfully with cross validation. However, when refitting, an unexpected error occurred. Shouldn't there be no issue since both fit and refit calls fit() in base.py and I have passed in the same parameters for both fit() and refit().

Ran the following code:
```python
automl = autosklearn.classification.AutoSklearnClassifier(
        tmp_folder='/tmp/autosklearn_cv_tmp',
        output_folder='/tmp/autosklearn_cv_out',
        delete_tmp_folder_after_terminate=False,
        resampling_strategy='cv',
        resampling_strategy_arguments={'folds': 10})

automl.fit(X_train.copy(), y_train.copy(), dataset_name='Emotion')

automl.refit(X_train.copy(), y_train.copy())

```

The relevant part of the stack trace:
```
/home/cs/anaconda3/lib/python3.6/site-packages/autosklearn/estimators.py in refit(self, X, y)
     48 
     49         """
---> 50         return self._automl.refit(X, y)
     51 
     52     def fit_ensemble(self, y, task=None, metric=None, precision='32',

/home/cs/anaconda3/lib/python3.6/site-packages/autosklearn/estimators.py in refit(self, X, y)
     48 
     49         """
---> 50         return self._automl.refit(X, y)
     51 
     52     def fit_ensemble(self, y, task=None, metric=None, precision='32',

/home/cs/anaconda3/lib/python3.6/site-packages/autosklearn/automl.py in refit(self, X, y)
    495                         with warnings.catch_warnings():
    496                             warnings.showwarning = send_warnings_to_log
--> 497                             model.fit(X.copy(), y.copy())
    498                         break
    499                     except ValueError as e:

/home/cs/anaconda3/lib/python3.6/site-packages/autosklearn/pipeline/base.py in fit(self, X, y, fit_params)
     86             a classification algorithm first.
     87         """
---> 88         X, fit_params = self.fit_transformer(X, y, fit_params=fit_params)
     89         self.fit_estimator(X, y, **fit_params)
     90         return self

/home/cs/anaconda3/lib/python3.6/site-packages/autosklearn/pipeline/classification.py in fit_transformer(self, X, y, fit_params)
     82 
     83     def fit_transformer(self, X, y, fit_params=None):
---> 84         self.num_targets = 1 if len(y.shape) == 1 else y.shape[1]
     85 
     86         if fit_params is None:

AttributeError: 'list' object has no attribute 'shape'

```

**Comment by mfeurer** [comment]
Thanks a lot for reporting this. I'm working on this issue, but it may take some time until the fix is in Auto-sklearn. An easy hotfix is that you cast `X` and `y` yourself prior to calling `refit(X, y)`.

**Comment by tanchangsheng** [comment]
Late but thank you for the recommendation!

**Comment by mkcedward** [comment]
@mfeurer Can you advise sample code for casting x and y ?

**Comment by mfeurer** [comment]
The appropriate cast should automatically happen in the next release. Fixed via #368.
